instance/tests/models/test_log_entry.py

```diff
@@ -134,6 +136,108 @@ def test_log_publish(self, mock_publish_data):
 
+    def test_log_delete(self):
+        """
+        Check `log_entries` output for combination of instance & server logs
+        """
+        server1 = self.server
+        server2 = OpenStackServerFactory(instance=self.instance, openstack_id='vm2_id')
+
+        self.instance.logger.info('Line #1, on instance')
+        server1.logger.info('Line #2, on server 1')
+        server2.logger.info('Line #3, on server 2')
+
+        self.assertEqual(LogEntry.objects.count(), 3)
+        # Delete server 1:
+        server1_id = server1.pk
+        server1.delete()
+        # Now its log entry should be deleted:
+        entries = LogEntry.objects.order_by('pk').all().values_list('text', flat=True)
+        for entry_text in entries:
+            self.assertNotIn('Line #2', entry_text)
+        self.assertIn('Line #1, on instance', entries[0])
+        self.assertIn('Line #3, on server 2', entries[1])
+        self.assertIn(
+            'Deleted 1 log entries for deleted open stack server instance with ID {}'.format(server1_id),
+            entries[2]
+        )
+
+    def test_log_num_queries(self):
+        """
+        Check that logging to the LogEntry table doesn't do more queries than necessary.
+
+        The expected queries upon inserting a log entry are:
+        1. SELECT (1) AS "a" FROM "django_content_type" WHERE "django_content_type"."id" = {content_type_id} LIMIT 1
+        2. SELECT (1) AS "a" FROM "instance_openstackserver" WHERE "instance_openstackserver"."id" = {object_id} LIMIT 1
+        3. INSERT INTO "instance_logentry" (...)
+
+        The first two are used to validate the foreign keys. 1. is added by django, and 2. is
+        added by us since the object_id foreign key constraint is not enforced by the database.
+        """
+        with self.assertNumQueries(3):
+            self.server.logger.info('some log message')
+
+    def test_log_delete_num_queries(self):
+        """
+        Check that the LogEntry.on_post_delete handler doesn't do more queries than necessary.
+        """
+        with self.assertNumQueries(2):  # one query to delete the server; one to delete the LogEntry
+            self.server.delete()
+        log_entry = LogEntry.objects.create(text='blah')
+        with self.assertNumQueries(1):
+            log_entry.delete()
+
+    def test_str_repr(self):
+        """
+        Test the string representation of a LogEntry object
+        """
+        msg = 'We have entered a spectacular binary star system in the Kavis Alpha sector'
+        with freeze_time("2015-10-20 20:10:15"):
+            self.server.logger.info(msg)
+        log_entry = LogEntry.objects.order_by('-pk')[0]
+        self.assertEqual(
+            str(log_entry),
+            '2015-10-20 20:10:15 |     INFO | instance.models.server    | instance=my.instance,server=vm1_id | ' + msg,
+        )
+
+    def test_invalid_content_type_object_id_combo(self):
+        """
+        Test that content_type cannot be set on its own
```

Comment:
@bradenmacdonald Nit: "Test that content_type and object_id cannot be set on their own"

I'm fine with updating this docstring as part of #63; it's definitely not a blocker to merging this PR :)